test/SlowTests/Client/TimeSeries/Query/TimeSeriesJavascriptProjections.cs

```diff
@@ -0,0 +1,2008 @@
+                    Assert.Equal(baseline.AddMonths(1).AddMinutes(61), result.Stocks.Results[1].Timestamp);
+
+                    Assert.Equal(new[] { 679d }, result.Stocks.Results[2].Values);
+                    Assert.Equal("tags/1", result.Stocks.Results[2].Tag);
+                    Assert.Equal(baseline.AddMonths(1).AddMinutes(62), result.Stocks.Results[2].Timestamp);
+
+
+                }
+            }
+        }
+
+        [Fact]
+        public void CanCallTimeSeriesDeclaredFunctionFromJavascriptDeclaredFunction()
+        {
+            using (var store = GetDocumentStore())
+            {
+                var baseline = DateTime.Today;
+
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new Person
+                    {
+                        Name = "Oren",
+                        Age = 60,
+                        WorksAt = "companies/1"
+                    }, "people/1");
+
+                    session.Store(new Company
+                    {
+                        Name = "HR"
+                    }, "companies/1");
+
+                    var tsf = session.TimeSeriesFor("people/1");
+
+                    tsf.Append("Heartrate", baseline.AddMinutes(61), "tags/1", new[] { 59d });
+                    tsf.Append("Heartrate", baseline.AddMinutes(62), "tags/2", new[] { 79d });
+                    tsf.Append("Heartrate", baseline.AddMinutes(63), "tags/2", new[] { 69d });
+
+                    tsf.Append("Heartrate", baseline.AddMonths(1).AddMinutes(61), "tags/1", new[] { 159d });
+                    tsf.Append("Heartrate", baseline.AddMonths(1).AddMinutes(62), "tags/1", new[] { 179d });
+                    tsf.Append("Heartrate", baseline.AddMonths(1).AddMinutes(63), "tags/2", new[] { 169d });
+
+                    tsf.Append("Stocks", baseline.AddMinutes(61), "tags/1", new[] { 559d });
+                    tsf.Append("Stocks", baseline.AddMinutes(62), "tags/2", new[] { 579d });
+                    tsf.Append("Stocks", baseline.AddMinutes(63), "tags/2", new[] { 569d });
+
+                    tsf.Append("Stocks", baseline.AddMonths(1).AddMinutes(61), "tags/1", new[] { 659d });
+                    tsf.Append("Stocks", baseline.AddMonths(1).AddMinutes(62), "tags/1", new[] { 679d });
+                    tsf.Append("Stocks", baseline.AddMonths(1).AddMinutes(63), "tags/2", new[] { 669d });
+
+                    session.SaveChanges();
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var rawQuery = session.Advanced.RawQuery<CustomRawQueryResult4>(
+@"declare timeseries ts(name) 
+{
+    from name between $start and $end
+    where Tag != 'tags/2'
+}
+declare function out(d) 
+{
+    var result = {};
+    var allTsNames = d['@metadata']['@timeseries'];
+    for (var i = 0; i < allTsNames.length; i++){
+        var name = allTsNames[i];
+        result[name] = ts(name);
```

Comment:
How can I pass different ranges here?

Reply:
```
declare timeseries ts(name, start, end) 
{
    from name between start and end
    where Tag != 'tags/2'
}
```
and then we can pass a diffrent start & end for each call 

added test